packages/core/src/performers/step-performer/StepPerformerPromptCreator.ts

```diff
@@ -151,9 +151,12 @@ export class StepPerformerPromptCreator {
             "```",
             previousStep.code,
             "```",
-            ...(previousStep.result
+            ...(previousStep.result && !previousStep.error
               ? [`- Result: ${previousStep.result}`]
               : []),
+            ...(index === previousSteps.length - 1 && previousStep.error
+                ? [`- Error:\n "${previousStep.error}" \nOccurred in your previous attempt. Try another approach for this step`]
+                : []),
```

Comment:
let's extract this to a method that handles the logic of `previousStepResultOrError(previousStep, isMostPreviousStep)`, and will return the expected string (result or error)